### ISSUE TITLE
ADD Behaviour when source is remote tar.gz URL is different from doc in 17.06.0-ce-win19 (12801)

### ISSUE THREAD

**Issue by tingasaurus** [moby/moby#34316]
**Description**

**Steps to reproduce the issue:**
1. Inside a Dockerfile, do something like
"ADD https://download.newrelic.com/server_monitor/release/newrelic-sysmond-${NEW_RELIC_VERSION}-linux.tar.gz /newrelic-sysmond.tar.gz
RUN tar xvfz /newrelic-sysmond.tar.gz && \
  rm /newrelic-sysmond.tar.gz"

**Describe the results you received:**
tar (child): /newrelic-sysmond.tar.gz: Cannot read: Is a directory

**Describe the results you expected:**
according to the doc
"Resources from remote URLs are not decompressed"

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:30:30 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:51:55 2017
 OS/Arch:      linux/amd64
 Experimental: true

```

**Output of `docker info`:**

```
Containers: 14
 Running: 0
 Paused: 0
 Stopped: 14
Images: 43
Server Version: 17.06.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.36-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.934GiB
Name: moby
ID: IMKS:JGI4:OJMQ:2PVQ:D2B4:SNAS:G54M:344D:W4NC:BPLU:N33Y:DJ6I
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 15
 Goroutines: 25
 System Time: 2017-07-31T03:35:53.5282438Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
windows 10, physical machine

**Comment by friism** [comment]
I think this is a [known problem](https://github.com/moby/moby/issues/33849) (also see the 17.06.0 release notes where this was published as a known errata) that's fixed in 17.06.1 and 17.07.0.

Could you try the Docker for Windows edge variant? It'll have a fixed Docker version.

cc @thaJeztah I think this can be closed as dupe.

**Comment by thaJeztah** [comment]
Thanks! Saw this reported, but didn't come round to replying yet. Yes this is a know issue and will be addressed in 17.06.1

I'll close, but feel free to discuss after that
